### DISCUSSION TITLE
Vercel部署出现问题

### DISCUSSION THREAD

**Discussion by SmallKunKun** [Qexo/Qexo#188] [Q&A]
15:29:43.781 | Cloning completed: 569.167ms
15:29:43.843 | Warning: Due to `builds` existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings
15:29:43.871 | Installing build runtime...
15:29:45.621 | Build runtime installed: 1.750s
15:29:45.647 | Installing build runtime...
15:29:48.148 | Build runtime installed: 2.500s
15:29:49.035 | No Build Cache available
15:29:49.674 | Collecting asgiref==3.4.1
15:29:49.742 | Downloading asgiref-3.4.1-py3-none-any.whl (25 kB)
15:29:49.782 | Collecting autopep8==1.5.7
15:29:49.796 | Downloading autopep8-1.5.7-py2.py3-none-any.whl (45 kB)
15:29:49.807 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.0/45.0 kB 4.3 MB/s eta 0:00:00
15:29:49.842 | Collecting beautifulsoup4==4.10.0
15:29:49.856 | Downloading beautifulsoup4-4.10.0-py3-none-any.whl (97 kB)
15:29:49.872 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.4/97.4 kB 6.2 MB/s eta 0:00:00
15:29:49.916 | Collecting boto==2.49.0
15:29:49.930 | Downloading boto-2.49.0-py2.py3-none-any.whl (1.4 MB)
15:29:49.975 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.4/1.4 MB 31.4 MB/s eta 0:00:00
15:29:50.456 | Collecting boto3==1.20.23
15:29:50.473 | Downloading boto3-1.20.23-py3-none-any.whl (131 kB)
15:29:50.481 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 131.8/131.8 kB 23.5 MB/s eta 0:00:00
15:29:51.045 | Collecting botocore==1.23.23
15:29:51.061 | Downloading botocore-1.23.23-py3-none-any.whl (8.4 MB)
15:29:51.150 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.4/8.4 MB 97.6 MB/s eta 0:00:00
15:29:51.219 | Collecting certifi==2021.5.30
15:29:51.236 | Downloading certifi-2021.5.30-py2.py3-none-any.whl (145 kB)
15:29:51.244 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 145.5/145.5 kB 23.2 MB/s eta 0:00:00
15:29:51.493 | Collecting cffi==1.15.0
15:29:51.510 | Downloading cffi-1.15.0-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (444 kB)
15:29:51.535 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 444.3/444.3 kB 19.1 MB/s eta 0:00:00
15:29:51.573 | Collecting charset-normalizer==2.0.4
15:29:51.596 | Downloading charset_normalizer-2.0.4-py3-none-any.whl (36 kB)
15:29:51.643 | Collecting click==8.0.1
15:29:51.658 | Downloading click-8.0.1-py3-none-any.whl (97 kB)
15:29:51.665 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.4/97.4 kB 20.8 MB/s eta 0:00:00
15:29:51.700 | Collecting colorama==0.4.4
15:29:51.714 | Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)
15:29:51.752 | Collecting Deprecated==1.2.13
15:29:51.767 | Downloading Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
15:29:51.795 | Collecting dj-database-url==0.5.0
15:29:51.809 | Downloading dj_database_url-0.5.0-py2.py3-none-any.whl (5.5 kB)
15:29:51.929 | Collecting Django==3.0.5
15:29:51.944 | Downloading Django-3.0.5-py3-none-any.whl (7.5 MB)
15:29:52.023 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.5/7.5 MB 99.3 MB/s eta 0:00:00
15:29:52.104 | Collecting djongo==1.3.6
15:29:52.120 | Downloading djongo-1.3.6.tar.gz (331 kB)
15:29:52.130 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 331.6/331.6 kB 42.6 MB/s eta 0:00:00
15:29:52.156 | Preparing metadata (setup.py): started
15:29:52.333 | Preparing metadata (setup.py): finished with status 'done'
15:29:52.379 | Collecting django-cors-headers==3.10.1
15:29:52.395 | Downloading django_cors_headers-3.10.1-py3-none-any.whl (12 kB)
15:29:52.428 | Collecting dnspython==1.16.0
15:29:52.466 | Downloading dnspython-1.16.0-py2.py3-none-any.whl (188 kB)
15:29:52.474 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 188.4/188.4 kB 32.5 MB/s eta 0:00:00
15:29:52.517 | Collecting gunicorn==20.1.0
15:29:52.531 | Downloading gunicorn-20.1.0-py3-none-any.whl (79 kB)
15:29:52.538 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.5/79.5 kB 15.9 MB/s eta 0:00:00
15:29:52.569 | Collecting idna==3.2
15:29:52.583 | Downloading idna-3.2-py3-none-any.whl (59 kB)
15:29:52.590 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.6/59.6 kB 11.7 MB/s eta 0:00:00
15:29:52.618 | Collecting jmespath==0.10.0
15:29:52.633 | Downloading jmespath-0.10.0-py2.py3-none-any.whl (24 kB)
15:29:52.669 | Collecting prettytable==2.2.0
15:29:52.686 | Downloading prettytable-2.2.0-py3-none-any.whl (23 kB)
15:29:52.741 | Collecting pyasn1==0.4.8
15:29:52.755 | Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
15:29:52.762 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 77.1/77.1 kB 15.5 MB/s eta 0:00:00
15:29:52.793 | Collecting pycodestyle==2.7.0
15:29:52.807 | Downloading pycodestyle-2.7.0-py2.py3-none-any.whl (41 kB)
15:29:52.813 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.7/41.7 kB 8.5 MB/s eta 0:00:00
15:29:52.840 | Collecting pycparser==2.21
15:29:52.853 | Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
15:29:52.860 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 kB 25.1 MB/s eta 0:00:00
15:29:52.911 | Collecting PyGithub==1.55
15:29:52.927 | Downloading PyGithub-1.55-py3-none-any.whl (291 kB)
15:29:52.937 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 291.7/291.7 kB 38.7 MB/s eta 0:00:00
15:29:52.976 | Collecting PyJWT==2.3.0
15:29:52.990 | Downloading PyJWT-2.3.0-py3-none-any.whl (16 kB)
15:29:53.448 | Collecting pymongo==3.12.0
15:29:53.466 | Downloading pymongo-3.12.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (531 kB)
15:29:53.478 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 531.9/531.9 kB 56.7 MB/s eta 0:00:00
15:29:53.530 | Collecting PyNaCl==1.4.0
15:29:53.548 | Downloading PyNaCl-1.4.0-cp35-abi3-manylinux1_x86_64.whl (961 kB)
15:29:53.563 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 961.1/961.1 kB 80.6 MB/s eta 0:00:00
15:29:53.599 | Collecting python-dateutil==2.8.2
15:29:53.613 | Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
15:29:53.621 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 247.7/247.7 kB 43.7 MB/s eta 0:00:00
15:29:53.647 | Collecting python-decouple==3.4
15:29:53.665 | Downloading python_decouple-3.4-py3-none-any.whl (9.5 kB)
15:29:53.753 | Collecting pytz==2021.1
15:29:53.767 | Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)
15:29:53.780 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 510.8/510.8 kB 54.5 MB/s eta 0:00:00
15:29:53.847 | Collecting requests==2.27.1
15:29:53.862 | Downloading requests-2.27.1-py2.py3-none-any.whl (63 kB)
15:29:53.869 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.1/63.1 kB 10.0 MB/s eta 0:00:00
15:29:53.904 | Collecting rsa==4.7.2
15:29:53.918 | Downloading rsa-4.7.2-py3-none-any.whl (34 kB)
15:29:53.963 | Collecting s3transfer==0.5.0
15:29:53.977 | Downloading s3transfer-0.5.0-py3-none-any.whl (79 kB)
15:29:53.985 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.2/79.2 kB 14.7 MB/s eta 0:00:00
15:29:54.016 | Collecting six==1.16.0
15:29:54.031 | Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
15:29:54.075 | Collecting soupsieve==2.2.1
15:29:54.092 | Downloading soupsieve-2.2.1-py3-none-any.whl (33 kB)
15:29:54.126 | Collecting sqlparse==0.2.4
15:29:54.140 | Downloading sqlparse-0.2.4-py2.py3-none-any.whl (38 kB)
15:29:54.168 | Collecting tcping==0.1.1rc1
15:29:54.184 | Downloading tcping-0.1.1rc1.tar.gz (4.1 kB)
15:29:54.190 | Preparing metadata (setup.py): started
15:29:54.363 | Preparing metadata (setup.py): finished with status 'done'
15:29:54.393 | Collecting toml==0.10.2
15:29:54.407 | Downloading toml-0.10.2-py2.py3-none-any.whl (16 kB)
15:29:54.516 | Collecting ujson==4.1.0
15:29:54.533 | Downloading ujson-4.1.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.whl (179 kB)
15:29:54.541 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.5/179.5 kB 33.4 MB/s eta 0:00:00
15:29:54.566 | Collecting Unipath==1.1
15:29:54.601 | Downloading Unipath-1.1.tar.gz (30 kB)
15:29:54.611 | Preparing metadata (setup.py): started
15:29:54.777 | Preparing metadata (setup.py): finished with status 'done'
15:29:54.838 | Collecting urllib3==1.26.7
15:29:54.855 | Downloading urllib3-1.26.7-py2.py3-none-any.whl (138 kB)
15:29:54.870 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 138.8/138.8 kB 10.7 MB/s eta 0:00:00
15:29:54.901 | Collecting wcwidth==0.2.5
15:29:54.915 | Downloading wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
15:29:54.962 | Collecting whitenoise==5.3.0
15:29:54.978 | Downloading whitenoise-5.3.0-py2.py3-none-any.whl (19 kB)
15:29:55.119 | Collecting wrapt==1.13.3
15:29:55.135 | Downloading wrapt-1.13.3-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (81 kB)
15:29:55.142 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 81.3/81.3 kB 13.8 MB/s eta 0:00:00
15:29:55.182 | Collecting Markdown==3.3.6
15:29:55.204 | Downloading Markdown-3.3.6-py3-none-any.whl (97 kB)
15:29:55.211 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.8/97.8 kB 18.0 MB/s eta 0:00:00
15:29:55.242 | Collecting html2text==2020.1.16
15:29:55.257 | Downloading html2text-2020.1.16-py3-none-any.whl (32 kB)
15:29:55.337 | Collecting pyyaml==6.0
15:29:55.354 | Downloading PyYAML-6.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (661 kB)
15:29:55.368 | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 661.8/661.8 kB 58.9 MB/s eta 0:00:00
15:29:55.670 | Requirement already satisfied: setuptools>=3.0 in /usr/local/lib/python3.9/site-packages (from gunicorn==20.1.0->-r requirements.txt (line 18)) (58.1.0)
15:29:56.141 | Collecting importlib-metadata>=4.4
15:29:56.155 | Downloading importlib_metadata-4.12.0-py3-none-any.whl (21 kB)
15:29:56.244 | Collecting zipp>=0.5
15:29:56.260 | Downloading zipp-3.8.1-py3-none-any.whl (5.6 kB)
15:29:56.338 | Using legacy 'setup.py install' for djongo, since package 'wheel' is not installed.
15:29:56.338 | Using legacy 'setup.py install' for tcping, since package 'wheel' is not installed.
15:29:56.338 | Using legacy 'setup.py install' for Unipath, since package 'wheel' is not installed.
15:29:56.544 | Installing collected packages: wcwidth, Unipath, sqlparse, pytz, python-decouple, pyasn1, dj-database-url, certifi, boto, zipp, wrapt, whitenoise, urllib3, ujson, toml, soupsieve, six, rsa, pyyaml, pymongo, PyJWT, pycparser, pycodestyle, prettytable, jmespath, idna, html2text, gunicorn, dnspython, colorama, click, charset-normalizer, asgiref, tcping, requests, python-dateutil, importlib-metadata, Django, Deprecated, cffi, beautifulsoup4, autopep8, PyNaCl, Markdown, djongo, django-cors-headers, botocore, s3transfer, PyGithub, boto3
15:29:56.593 | Running setup.py install for Unipath: started
15:29:56.794 | Running setup.py install for Unipath: finished with status 'done'
15:29:58.707 | Running setup.py install for tcping: started
15:29:58.914 | Running setup.py install for tcping: finished with status 'done'
15:30:00.918 | Running setup.py install for djongo: started
15:30:01.180 | Running setup.py install for djongo: finished with status 'done'
15:30:02.054 | Successfully installed Deprecated-1.2.13 Django-3.0.5 Markdown-3.3.6 PyGithub-1.55 PyJWT-2.3.0 PyNaCl-1.4.0 Unipath-1.1 asgiref-3.4.1 autopep8-1.5.7 beautifulsoup4-4.10.0 boto-2.49.0 boto3-1.20.23 botocore-1.23.23 certifi-2021.5.30 cffi-1.15.0 charset-normalizer-2.0.4 click-8.0.1 colorama-0.4.4 dj-database-url-0.5.0 django-cors-headers-3.10.1 djongo-1.3.6 dnspython-1.16.0 gunicorn-20.1.0 html2text-2020.1.16 idna-3.2 importlib-metadata-4.12.0 jmespath-0.10.0 prettytable-2.2.0 pyasn1-0.4.8 pycodestyle-2.7.0 pycparser-2.21 pymongo-3.12.0 python-dateutil-2.8.2 python-decouple-3.4 pytz-2021.1 pyyaml-6.0 requests-2.27.1 rsa-4.7.2 s3transfer-0.5.0 six-1.16.0 soupsieve-2.2.1 sqlparse-0.2.4 tcping-0.1.1rc1 toml-0.10.2 ujson-4.1.0 urllib3-1.26.7 wcwidth-0.2.5 whitenoise-5.3.0 wrapt-1.13.3 zipp-3.8.1
15:30:02.054 | WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
15:30:02.178 |  
15:30:02.179 | [notice] A new release of pip available: 22.1.2 -> 22.2
15:30:02.179 | [notice] To update, run: pip install --upgrade pip
15:31:03.367 | Traceback (most recent call last):
15:31:03.368 | File "/vercel/path1/manage.py", line 22, in <module>
15:31:03.368 | main()
15:31:03.368 | File "/vercel/path1/manage.py", line 18, in main
15:31:03.368 | execute_from_command_line(sys.argv)
15:31:03.368 | File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
15:31:03.368 | utility.execute()
15:31:03.368 | File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 395, in execute
15:31:03.368 | self.fetch_command(subcommand).run_from_argv(self.argv)
15:31:03.368 | File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 328, in run_from_argv
15:31:03.368 | self.execute(*args, **cmd_options)
15:31:03.368 | File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 369, in execute
15:31:03.368 | output = self.handle(*args, **options)
15:31:03.368 | File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 83, in wrapped
15:31:03.368 | res = handle_func(*args, **kwargs)
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/makemigrations.py", line 101, in handle
15:31:03.369 | loader.check_consistent_history(connection)
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/loader.py", line 283, in check_consistent_history
15:31:03.369 | applied = recorder.applied_migrations()
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/recorder.py", line 76, in applied_migrations
15:31:03.369 | if self.has_table():
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/recorder.py", line 56, in has_table
15:31:03.369 | return self.Migration._meta.db_table in self.connection.introspection.table_names(self.connection.cursor())
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/introspection.py", line 48, in table_names
15:31:03.369 | return get_names(cursor)
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/introspection.py", line 43, in get_names
15:31:03.369 | return sorted(ti.name for ti in self.get_table_list(cursor)
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/djongo/introspection.py", line 47, in get_table_list
15:31:03.369 | for c in cursor.db_conn.list_collection_names()
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/pymongo/database.py", line 880, in list_collection_names
15:31:03.369 | for result in self.list_collections(session=session, **kwargs)]
15:31:03.369 | File "/usr/local/lib/python3.9/site-packages/pymongo/database.py", line 842, in list_collections
15:31:03.370 | return self.__client._retryable_read(
15:31:03.370 | File "/usr/local/lib/python3.9/site-packages/pymongo/mongo_client.py", line 1514, in _retryable_read
15:31:03.370 | server = self._select_server(
15:31:03.370 | File "/usr/local/lib/python3.9/site-packages/pymongo/mongo_client.py", line 1346, in _select_server
15:31:03.370 | server = topology.select_server(server_selector)
15:31:03.370 | File "/usr/local/lib/python3.9/site-packages/pymongo/topology.py", line 244, in select_server
15:31:03.370 | return random.choice(self.select_servers(selector,
15:31:03.370 | File "/usr/local/lib/python3.9/site-packages/pymongo/topology.py", line 202, in select_servers
15:31:03.371 | server_descriptions = self._select_servers_loop(
15:31:03.371 | File "/usr/local/lib/python3.9/site-packages/pymongo/topology.py", line 218, in _select_servers_loop
15:31:03.371 | raise ServerSelectionTimeoutError(
15:31:03.371 | pymongo.errors.ServerSelectionTimeoutError: connection closed,connection closed,connection closed, Timeout: 30s, Topology Description: <TopologyDescription id: 62dcf4faadcea79145e1bd7f, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('cluster0-shard-00-00.0furp.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('connection closed')>, <ServerDescription ('cluster0-shard-00-01.0furp.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('connection closed')>, <ServerDescription ('cluster0-shard-00-02.0furp.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('connection closed')>]>
15:32:04.421 | Traceback (most recent call last):
15:32:04.422 | File "/vercel/path1/manage.py", line 22, in <module>
15:32:04.422 | main()
15:32:04.422 | File "/vercel/path1/manage.py", line 18, in main
15:32:04.422 | execute_from_command_line(sys.argv)
15:32:04.422 | File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
15:32:04.422 | utility.execute()
15:32:04.422 | File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 395, in execute
15:32:04.422 | self.fetch_command(subcommand).run_from_argv(self.argv)
15:32:04.422 | File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 328, in run_from_argv
15:32:04.422 | self.execute(*args, **cmd_options)
15:32:04.422 | File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 369, in execute
15:32:04.424 | output = self.handle(*args, **options)
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 83, in wrapped
15:32:04.425 | res = handle_func(*args, **kwargs)
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/migrate.py", line 86, in handle
15:32:04.425 | executor = MigrationExecutor(connection, self.migration_progress_callback)
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/executor.py", line 18, in __init__
15:32:04.425 | self.loader = MigrationLoader(self.connection)
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/loader.py", line 49, in __init__
15:32:04.425 | self.build_graph()
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/loader.py", line 212, in build_graph
15:32:04.425 | self.applied_migrations = recorder.applied_migrations()
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/recorder.py", line 76, in applied_migrations
15:32:04.425 | if self.has_table():
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/db/migrations/recorder.py", line 56, in has_table
15:32:04.425 | return self.Migration._meta.db_table in self.connection.introspection.table_names(self.connection.cursor())
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/introspection.py", line 48, in table_names
15:32:04.425 | return get_names(cursor)
15:32:04.425 | File "/usr/local/lib/python3.9/site-packages/django/db/backends/base/introspection.py", line 43, in get_names
15:32:04.425 | return sorted(ti.name for ti in self.get_table_list(cursor)
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/djongo/introspection.py", line 47, in get_table_list
15:32:04.426 | for c in cursor.db_conn.list_collection_names()
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/pymongo/database.py", line 880, in list_collection_names
15:32:04.426 | for result in self.list_collections(session=session, **kwargs)]
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/pymongo/database.py", line 842, in list_collections
15:32:04.426 | return self.__client._retryable_read(
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/pymongo/mongo_client.py", line 1514, in _retryable_read
15:32:04.426 | server = self._select_server(
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/pymongo/mongo_client.py", line 1346, in _select_server
15:32:04.426 | server = topology.select_server(server_selector)
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/pymongo/topology.py", line 244, in select_server
15:32:04.426 | return random.choice(self.select_servers(selector,
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/pymongo/topology.py", line 202, in select_servers
15:32:04.426 | server_descriptions = self._select_servers_loop(
15:32:04.426 | File "/usr/local/lib/python3.9/site-packages/pymongo/topology.py", line 218, in _select_servers_loop
15:32:04.426 | raise ServerSelectionTimeoutError(
15:32:04.426 | pymongo.errors.ServerSelectionTimeoutError: connection closed,connection closed,connection closed, Timeout: 30s, Topology Description: <TopologyDescription id: 62dcf5383846ed2cfc7621a7, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('cluster0-shard-00-00.0furp.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('connection closed')>, <ServerDescription ('cluster0-shard-00-01.0furp.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('connection closed')>, <ServerDescription ('cluster0-shard-00-02.0furp.mongodb.net', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('connection closed')>]>
15:32:05.000 | Error: Command "./migrate.sh" exited with 1

**Comment by am-abudu** [comment]
检查数据库配置，并且是否是按照文档要求的使用AWS / N. Virginia (us-east-1)

**Reply by SmallKunKun** [reply]
![image](https://user-images.githubusercontent.com/54430298/180638082-6e257ece-d42b-4260-8ba7-b93baf4cb1d9.png)

**Reply by am-abudu** [reply]
请回复环境变量配置（密码打码）

**Comment by SmallKunKun** [comment]
![image](https://user-images.githubusercontent.com/54430298/180638155-15950fac-d882-4ee9-ab0b-c8cd4736d485.png)

**Reply by am-abudu** [reply]
数据库是否配置IP白名单？

**Reply by SmallKunKun** [reply]
没有 我想到了 但第一次使用mongodb不知道如何操作 而且将什么ip添加呢

**Reply by am-abudu** [reply]
参考文档`允许所有 IP 地址的连接`

**Reply by SmallKunKun** [reply]
pymongo.errors.OperationFailure: bad auth : Authentication failed., full error: {'ok': 0, 'errmsg': 'bad auth : Authentication failed.', 'code': 8000, 'codeName': 'AtlasError'}
--
16:15:42.468 | Error: Command "./migrate.sh" exited with 1

**Reply by am-abudu** [reply]
数据库名、账户密码错误，请自行纠错，检查是否有无关空格之类的信息

**Reply by SmallKunKun** [reply]
部署好了 谢谢
